src/sentry/static/sentry/app/utils/integrationUtil.tsx

```diff
@@ -62,20 +67,18 @@ export type SingleIntegrationEvent = {
     | 'Integrations: Integration Tab Clicked'
     | 'Integrations: Config Saved'
     | 'Integrations: Plugin Add to Project Clicked'
-    | 'Integrations: Upgrade Plan Modal Opened';
+    | 'Integrations: Upgrade Plan Modal Opened'
+    | 'Integrations: Resolve Now Clicked'
+    | 'Integrations: Reauth Start'
+    | 'Integrations: Reauth Complete';
   integration: string; //the slug
+  integration_type: 'plugin' | 'first_party' | 'sentry_app' | 'document_integration';
   already_installed?: boolean;
   integration_tab?: 'configurations' | 'overview';
   plan?: string;
-} & (SentryAppEvent | NonSentryAppEvent);
-
-type SentryAppEvent = {
-  integration_type: 'sentry_app';
   //include the status since people might do weird things testing unpublished integrations
-  integration_status: SentryAppStatus;
-};
-type NonSentryAppEvent = {
```

Comment:
I had to simplify some of the types for analytics events and just make `integration_status` an optional parameter.